src/main/java/Duke.java

```diff
@@ -1,10 +1,358 @@
+            throw new TasketException("The task index must be a number");
+        } catch (IndexOutOfBoundsException e) {
+            throw new TasketException("The task index exceeds the size of list");
+        }
+    }
+
+    public static void rewriteSaveFile() throws TasketException {
+        FileWriter fw = null;
+
+        try {
+            fw = new FileWriter(dataFile, false);
+            fw.write("");
+
+            for (Task task : tasks) {
+                fw.append(task.exportSaveFormat()).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new TasketException("IO Error");
+        } finally {
+            try {
+                if (fw != null) {
+                    fw.close();
+                }
+            } catch (IOException e) {
+                throw new TasketException("IO Error");
+            }
+        }
+    }
+
+    public static void retrieveSavedTasks() throws TasketException {
+        Scanner sc = null;
+        try {
+            sc = new Scanner(dataFile);
+
+            while (sc.hasNext()) {
+                parseLine(sc.nextLine());
+            }
+        } catch (FileNotFoundException e) {
+            throw new TasketException("File not found");
+        } finally {
+            if (sc != null) {
+                sc.close();
+            }
+        }
+    }
+
+    public static void parseLine(String savedTask) {
+        String[] taskElements = savedTask.split(" \\| ");
+        try {
+            Task generatedTask = null;
+            switch (taskElements[0]) {
+            case "T":
+                if (taskElements.length != 3) {
+                    throw new TasketException("Invalid task format");
+                }
+                generatedTask = new ToDo(taskElements[2]);
+                break;
+            case "D":
+                if (taskElements.length != 4) {
+                    throw new TasketException("Invalid task format");
+                }
+                generatedTask = new Deadline(taskElements[2], taskElements[3]);
+                break;
+            case "E":
+                if (taskElements.length != 5) {
+                    throw new TasketException("Invalid task format");
+                }
+                generatedTask = new Event(taskElements[2], taskElements[3], taskElements[4]);
+                break;
+            default:
+                throw new TasketException("Invalid task type");
+            }
+
+            if (taskElements[1].equals("1")) {
+                generatedTask.markAsDone();
+            }
+
+            tasks.add(generatedTask);
+
+        } catch (TasketException e) {
+            showErrorText(e.getMessage());
+        }
+    }
+
+    public static void parseInput(String prompt) throws TasketException {
+        String action = prompt.split(" ", 2)[0];
+
+        switch (action) {
+        case "todo":
+        case "deadline":
+        case "event":
```

Comment:
The explicit `//Fallthrough` comment should be included whenever there is a case statement without a break statement.